src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Usage/ImplementGenericMathInterfacesCorrectlyTests.cs

```diff
@@ -0,0 +1,315 @@
+{
+    public static Test Parse(ReadOnlySpan<char> s, IFormatProvider provider)
+    {
+        throw new NotImplementedException();
+    }
+
+    public static bool TryParse(ReadOnlySpan<char> s, IFormatProvider provider, out Test result)
+    {
+        throw new NotImplementedException();
+    }
+
+    public static bool TryParse(string s, IFormatProvider provider, out Test result)
+    {
+        throw new NotImplementedException();
+    }
+
+    static Test IParsable<Test>.Parse(string s, IFormatProvider provider)
+    {
+        throw new NotImplementedException();
+    }
+}
+
+public readonly struct MyDate<TSelf> : IParsable<TSelf> where TSelf : IParsable<TSelf>
+{
+    public static TSelf Parse(string s, IFormatProvider provider)
+    {
+        throw new NotImplementedException();
+    }
+
+    public static bool TryParse(string s, IFormatProvider provider, out TSelf result)
+    {
+        throw new NotImplementedException();
+    }
+}").RunAsync();
+        }
+
+        [Fact]
+        public async Task ISpanParsableNotImplementedCorrectly()
+        {
+            await PopulateTestCs(@"
+using System;
+
+public class Test : ISpanParsable<{|#0:DateOnly|}> // 'ISpanParsable' interface requires the derived type 'Test' used for the 'TSelf' type parameter
+{
+    public static DateOnly Parse(ReadOnlySpan<char> s, IFormatProvider provider)
+    {
+        throw new NotImplementedException();
+    }
+
+    public static bool TryParse(ReadOnlySpan<char> s, IFormatProvider provider, out DateOnly result)
+    {
+        throw new NotImplementedException();
+    }
+
+    public static bool TryParse(string s, IFormatProvider provider, out DateOnly result)
+    {
+        throw new NotImplementedException();
+    }
+
+    static DateOnly IParsable<DateOnly>.Parse(string s, IFormatProvider provider)
+    {
+        throw new NotImplementedException();
+    }
+}", VerifyCS.Diagnostic(ImplementGenericMathInterfacesCorrectly.GMInterfacesRule).WithLocation(0).WithArguments("ISpanParsable", "Test", "TSelf")).RunAsync();
+        }
+
+        [Fact]
+        public async Task IAdditionOperatorsNotImplementedCorrectly()
+        {
+            await PopulateTestCs(@"
+using System;
+using System.Numerics;
+
+public class Test : IAdditionOperators<Test, MyTest, long>
+{
+    public static long operator +(Test left, MyTest right)
+    {
+        throw new NotImplementedException();
+    }
+
+    public static long operator checked +(Test left, MyTest right)
+    {
+        throw new NotImplementedException();
+    }
+}
+public class MyTest : IAdditionOperators<[|Test|], MyTest, long> // 'IAdditionOperators' interface requires the derived type 'MyTest' used for the 'TSelf' type parameter
+{    public static long operator +(Test left, MyTest right)
+    {
+        throw new NotImplementedException();
+    }
+
+    public static long operator checked +(Test left, MyTest right)
+    {
+        throw new NotImplementedException();
+    }
+}").RunAsync();
```

Comment:
Doesn't this one need a diagnostic?

I see that `IAdditionOperators` doesn't inherit from `IParsable`.

Reply:
It has a diagnostic at row 160, the markup syntax `[|...|]` shows the location, just not checking the exact diagnostic and parameters, most of the test checking that though

Reply:
> I see that IAdditionOperators doesn't inherit from IParsable.

Doesn't matter, `IAdditionOperators` is included in the list